operator/builtin/transformer/remove/rootable_field.go

```diff
@@ -0,0 +1,147 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package remove
+
+import (
+	"encoding/json"
+
+	"github.com/open-telemetry/opentelemetry-log-collection/entry"
+)
+
+// RootableField represents a potential field on an entry.
+// It differs from a normal Field in that it allows users to
+// specify `$resource` or `$attributes` with the intention
+// of referring to "all" fields within those groups.
+// It is used to get, set, and delete values at this field.
+// It is deserialized from JSON dot notation.
+type RootableField struct {
```

Comment:
I think we should probably make this type private, since it has some special behaviors that only make sense for the `remove` operator. 

We may eventually need to generalize it, but for now, let's not imply that it's ready for use outside of this operator.

Reply:
bump

Reply:
@Mrod1598 This may have gotten buried.